### PR TITLE
Fix build on 10.4 and i386

### PR DESCRIPTION
**Pull request by mistydemeo**
This adapts a few fixes already in use on Tigerbrew.

### CODE REVIEW

File: Formula/portable-openssl.rb [hunk 1]
```diff
@@ -37,14 +42,18 @@ def arch_args
...
+    args = %W[
       --prefix=#{prefix}
       --openssldir=#{openssldir}
       no-ssl2
       zlib-dynamic
       no-shared
       enable-cms
     ]
+
+    args << "no-asm" if MacOS.version < :leopard
```
Comment (mistydemeo) [inline Formula/portable-openssl.rb]: Xcode 2.5's assembler is too old, and I'm not yet at the point of backporting a new assembler.

File: Formula/portable-openssl.rb [hunk 2]
```diff
@@ -16,6 +16,11 @@ class PortableOpenssl < PortableFormula
     sha256 "0f119da204025da7808273fab42ed8e030cafb5c7ea4e1deda4e75f066f528fb"
   end
 
+  patch :p0 do
```
Comment (mistydemeo) [inline Formula/portable-openssl.rb]: Necessary for i386 builds.

File: Formula/portable-openssl.rb [hunk 3]
```diff
@@ -74,7 +83,7 @@ def install
       system "make"
       system "make", "test"
 
-      if build.with? "universal"
+      if build.with? "universal" && archs.length > 1
```
Comment (MikeMcQuaid) [inline Formula/portable-openssl.rb]: Probably needs parens here to work as expected.

File: Formula/portable-openssl.rb [hunk 4]
```diff
@@ -83,7 +92,7 @@ def install
     system "make", "install", "MANDIR=#{man}"
     rm_rf man
 
-    if build.with? "universal"
+    if build.with? "universal" && archs.length > 1
```
Comment (MikeMcQuaid) [inline Formula/portable-openssl.rb]: Probably needs parens here to work as expected.

File: Formula/portable-openssl.rb (hunk 3 above)
Comment (mistydemeo) [inline Formula/portable-openssl.rb]: Oh, good point.

I'm also rethinking my universal approach - even if "universal" usually doesn't include 64-bit archs on Tiger and Leopard, I may look at making the portable formulae build i386/ppc.

Comment (xu-cheng) [inline Formula/portable-openssl.rb]: Just my two cents, this could be just `if archs.length > 1`

**Comment by mistydemeo** [comment]
Removed the universal commit.

**Pull request merged**